src/frontend/web_application/src/scenes/Contact/components/PublicKeyList/presenter.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Trans } from '@lingui/react';
+import { Button, Icon } from '../../../../components';
+import PublicKeyForm from '../PublicKeyForm';
+
+import './style.scss';
+
+const KEY_QUALITY_CLASSES = ['weak', 'average', 'good'];
+const KEY_QUALITY_ICONS = ['exclamation-triangle', 'expire-soon', 'info-circle'];
+
+class PublicKeyList extends Component {
+  static propTypes = {
+    contactId: PropTypes.string.isRequired,
+    publicKeys: PropTypes.arrayOf(PropTypes.shape({})),
+    requestPublicKeys: PropTypes.func.isRequired,
+    didInvalidate: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    didInvalidate: false,
+    publicKeys: [],
+  };
+
+  state = {
+    editMode: false,
+  };
+
+  componentDidMount() {
+    const {
+      publicKeys, contactId, requestPublicKeys,
+    } = this.props;
+
+    if (publicKeys.length === 0 || this.state.didInvalidate) {
+      requestPublicKeys({ contactId });
+    }
+  }
+
+  componentDidUpdate() {
+    const {
+      publicKeys, contactId, requestPublicKeys,
+    } = this.props;
+
+    if (publicKeys.length === 0 || this.state.didInvalidate) {
+      requestPublicKeys({ contactId });
+    }
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    return { ...state, didInvalidate: props.didInvalidate };
```

Comment:
I don't like this method: 

> This method exists for rare use cases

cf. https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

I translate it like this: "don't use this lifecycle method because it will break your application and it kills kitten."

A bad usage may result in an infinite render loop.

--- 

I might understand what you tried (I've made this error too):

I think you don't need `didInvalidate` in state at all.

```
componentDidMount() {
  if ((emptyPubKeys || this.props.didInvalidate) && !pubKeysRequestIsRunning) {
    requestPubKeys()
  }
}

// no componentDidUpdate()

onSuccess = () => {
  this.setState({ editMode: false }, () => {
    requestPubKeys();
  });
}
```